### PR TITLE
bpo-14156:Add argparse.FileType for a mode of 'rb' and 'wb'

### PR DESCRIPTION
**Pull request by hongweipeng**
Change patch to a RP.

https://bugs.python.org/issue14156

### CODE REVIEW

File: Lib/argparse.py [hunk 1]
```diff
@@ -1186,17 +1186,21 @@ def __call__(self, string):
...
+                file = _sys.stdout.fileno()
             else:
                 msg = _('argument "-" with mode %r') % self._mode
                 raise ValueError(msg)
+            closefd = False
+        else:
+            file = string
+            closefd = True
 
         # all other arguments are used as file names
```
Comment (serhiy-storchaka) [inline Lib/argparse.py]: Shouldn't this comment be moved 3 lines above?

File: Lib/test/test_argparse.py [hunk 2]
```diff
@@ -1476,6 +1481,17 @@ def __eq__(self, other):
             text = text.decode('ascii')
         return self.name == other.name == text
 
+class FilenoMode(object):
```
Comment (serhiy-storchaka) [inline Lib/test/test_argparse.py:1484]: Inheriting `object` is redundant.

Comment (hongweipeng) [reply Lib/test/test_argparse.py:1484]: It doesn't matter, it is  similar to `class RFile(object)` in the file.

Comment (hongweipeng) [reply Lib/test/test_argparse.py:1484]: Do I need to modify it?

**Pull request closed without merging**